src/sunmemory/sycl/sundials_sycl_memory.cpp

```diff
@@ -203,26 +199,23 @@ int SUNMemoryHelper_Alloc_Sycl(SUNMemoryHelper helper, SUNMemory* memptr,
     SUNDIALS_DEBUG_PRINT(
       "ERROR in SUNMemoryHelper_Alloc_Sycl: unknown memory type\n");
     free(mem);
-    return -1;
+    return SUN_ERR_CORRUPT;
```

Comment:
You'd used `SUN_ERR_OUTOFRANGE` in the CUDA version, but I think the most appropriate error code is
```suggestion
    return SUN_ERR_INCOMPATIBLE;
```